nucleus/deployment/common/src/main/java/org/glassfish/deployment/common/GenericAnnotationDetector.java

```diff
@@ -151,8 +152,7 @@ public void scanArchive(ReadableArchive archive) {
                     } finally {
                         is.close();
                     }
-                } else if (entryName.endsWith(".jar") && 
-                    entryName.indexOf('/') == -1) {
+                } else if (!entryName.contains("/")) {
```

Comment:
because otherwise it will not scan non-jars for annotations. The code below it is in a try statement anyway so any problems will be caught.